### ISSUE TITLE
Email template displayed incorrectly when using depend inside an if

### ISSUE THREAD

**Issue by igi8819** [magento/magento2#36438]
### Preconditions and environment

- Magento version: 2.4.5-p1
- Magento version: 2.4.4-p2

### Steps to reproduce

1. Go to admin -> marketing -> email templates
2. Click Add new email template
3. Load Default Template form "New Order" 
4.
Add these lines before greeting/or wherever you want
`{{if order_id}}
                {{depend order_id}}
                {{trans 'My order id is: <span class="no-link">%order_id</span>.' order_id=$order_id |raw}}
                {{/depend}}
            {{/if}}`

So looks like this:
<table>
    <tr class="email-intro">
        <td>

            {{if order_id}}
                {{depend order_id}}
                {{trans 'My order id is: <span class="no-link">%order_id</span>.' order_id=$order_id |raw}}
                {{/depend}}
            {{/if}}

            <p class="greeting">{{trans "%customer_name," customer_name=$order_data.customer_name}}</p>

5. Save email template and go to stores -> configuration -> sales -> sales emails
6. Add this new email template to "New Order Confirmation Template" than save it
7. Clear cache
8. Place and order in frontend and see the email content.
![image](https://user-images.githubusercontent.com/10732781/200611099-6f13049d-3272-484d-8a45-17eb7e16b06b.png)

The if condition is not processed.

I see that this in 2.4.5 (without p2) it is working.
Probably the issue is in the filter function, which is in the vendor/magento/framework/Filter/Template.php file

When i compare this with 2.4.5 version it way to different

**2.4.5-p2**
![image](https://user-images.githubusercontent.com/10732781/200611893-e49a0429-6077-4909-9d94-99b4c682804e.png)

**2.4.5**
![image](https://user-images.githubusercontent.com/10732781/200612004-c5b9bc80-44de-4034-9e24-991100786e53.png)

### Expected result

if condition in email template processed correctly

### Actual result

if condition in email template not processed correctly

### Additional information

this issue also happening on Magento 2.4.4-p2

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @igi8819,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop and 2.4.5 instances and the issue is not reproducible on both the instances.**Email Template Displayed Correctly**.Kindly refer the screenshots.

**Steps to reproduce** :

1.Go to admin -> marketing -> email templates
2.Click Add new email template
3.Load Default Template form "New Order"
4.Add these lines before greeting/or wherever you want
{{if order_id}} {{depend order_id}} {{trans 'My order id is: <span class="no-link">%order_id</span>.' order_id=$order_id |raw}} {{/depend}} {{/if}}.
5.Save email template and go to stores -> configuration -> sales -> sales emails
6.Add this new email template to "New Order Confirmation Template" than save it
7.Clear cache
8.Place and order in frontend and see the email content.


<img width="1086" alt="Screenshot 2022-11-09 at 2 38 39 PM" src="https://user-images.githubusercontent.com/51680745/200791511-3652ab14-21cb-42b5-868c-ce5b8479d051.png">



Kindly recheck the behavior on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by igi8819** [comment]
I mentioned that in **2.4.5** and **2.4.4** is **WORKING**!

**Not Working in:** 
- 2.4.5-p1
- 2.4.4-p2

So please test in 2.4.5-p1 and 2.4.4-p2 (security version)

**Comment by engcom-Bravo** [comment]
Hi @igi8819,

Thanks for your Update.

As per steps to reproduce the issue is reproducible in 2.4.5-p1 and 2.4.4-p2, but it is recommended to verify the issue on Magento 2.4-develop instance as it is an upcoming 2.4.x release and have latest code base.

we can accept the issues if they are reproducible in 2.4-develop branch.

Thanks.

**Comment by santibm** [comment]
@engcom-Bravo how are we then supposed to deal with this, an issue is not an issue because is not on the develop branch?! Then how come a security patch is released and promoted but it ends up creating another issue and can't be considered an issue!?

@igi8819 did you come up with any idea on how to deal with this?! 

Thanks

**Comment by igi8819** [comment]
@santibm i did not wait for fix, i just created a composer patch in my project and i updated the file from 2.4.4
My project is on 2.4.4-p2.
The files is: vendor/magento/framework/Filter/Template.php

**Comment by hostep** [comment]
@nathanjosiah: what's the reason why the recent security fixes haven't been merged yet in the `2.4-develop` branch? It's a bit unfortunate because that makes it hard for these engcom guys here to verify this issue.

**Comment by nathanjosiah** [comment]
@magento export issue to Jira project AC as Bug

**Comment by nathanjosiah** [comment]
@hostep None of the 2.4.6 security fixes will be merged back into `2.4-develop` until after 2.4.6 is GA in March'23. `2.4.5-p1`, `2.4.4-p2` and `2.4.6` all have their own scope and all security fixes are kept private until the release is GA.

As for the issue being described here, we are taking a look at this issue via internal ticket AC-7093

**Comment by hostep** [comment]
> @hostep None of the 2.4.6 security fixes will be merged back into `2.4-develop` until after 2.4.6 is GA in March'23. `2.4.5-p1`, `2.4.4-p2` and `2.4.6` all have their own scope and all security fixes are kept private until the release is GA.
> 
> As for the issue being described here, we are taking a look at this issue via internal ticket AC-7093

@nathanjosiah: I'm talking about the security fixes from 2.4.5-p1 (which already went GA more than a month ago), not upcoming ones from 2.4.6 😉 
These really should get merged in 2.4-develop, otherwise bugs introduced by them can't be verified.

**Comment by engcom-Lima** [comment]
Hi @igi8819 ,

Thanks for your contribution and collaboration.

We have tried to reproduce the issue in latest **2.4-develop** but issue is not reproducible. This issue will get fixed in upcoming version. Hence , closing this issue as not reproducible. Please feel free to reopen this ticket if you are facing any issue.

Thanks

**Comment by nathanjosiah** [comment]
@hostep The ticket number is AC-7896 but it hasn't been merged yet. We expect it to be merged to `2.4-develop` soon.

**Comment by nathanjosiah** [comment]
That said, I'm not sure why this was closed by @engcom-Lima. This is a valid issue which has been escalated by a number of merchants. However, there is also workaround of just not using this specific combination. You can recreate the same templates by using a combination of `if` and `else` instead of depend.

**Comment by nathanjosiah** [comment]
We are going to get the fix moved to the public branches.

**Comment by nathanjosiah** [comment]
Small update on this, the changes from AC-7093 are based on some security work that was done for the upcoming GA on March 14. For that reason, we cannot move the change to the public branches until the final release branches for 2.4.6 (which contain security fixes) are synced back into `2.4-develop`. 
That means it will be at LEAST March 14 before we can move them but this process of synchronization usually takes a couple of weeks to finalize for every GA release.

**Comment by rav-redchamps** [comment]
Reproducible in **2.4.6-p3** as well.

**Comment by adamzero1** [comment]
For anyone else who is getting this issue on 2.4.6-p3, this is a patch pulling in the lastest version of `vendor/magento/framework/Filter/Template.php` (from `2.4-develop`)
This may also work on other versions, but I haven't tested.

1. Install the patch. (Alternatively download and store in local path)
  ```bash
  composer config --json --merge extra.patches.magento/framework '{"Fixes https://github.com/magento/magento2/issues/36438":"https://raw.githubusercontent.com/zero1limited/magento2- patches/master/patches/issue/36438/2.4.6-p3.patch"}'
  ```
2. Apply the patch
  ```bash
  composer install
  ```
